### PR TITLE
Add generic error modal pop-ups for library and keep actions.

### PR DESCRIPTION
**Pull request by yipingliao**
@andrewconner @ahsu1230 
https://app.asana.com/0/17575799252940/19685910579384

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/keep/keep.js [hunk 1]
```diff
@@ -129,22 +129,6 @@ angular.module('kifi')
           keep.descHtml = formatDesc(keep.siteName || keep.url);
         }
 
-        function keepOne (keep) {
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/keep/keep.js:132]: I cleaned up some unused code in this file.

Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/keep/keep.js:132]: :+1:

File: kifi-backend/modules/shoebox/angular/src/keeps/keepActionService.js [hunk 2]
```diff
@@ -214,17 +214,6 @@ angular.module('kifi')
       });
     }
 
-    function togglePrivateMany(keeps) {
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/keeps/keepActionService.js:217]: More clean-up: no more toggling public and private keeps.

File: kifi-backend/modules/shoebox/angular/src/libraries/manageLibrary.js [hunk 3]
```diff
@@ -81,21 +81,18 @@ angular.module('kifi')
 
             var error = err.data && err.data.error;
             switch (error) {
-              case 'library name already exists for user':  // deprecated
-              case 'library_name_exists':
-                scope.$error.general = 'You already have a library with this name. Pick another.';
-                break;
-              case 'invalid library name':  // deprecated
               case 'invalid_name':
+                scope.$error.general = 'The name you picked isn\'t valid. Try using only letters and numbers.';
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/libraries/manageLibrary.js:85]: @ahsu1230 I updated these so that they are current.

**Comment by yipingliao** [comment]
@andrewconner Aaron and I are debating whether this pattern of catching errors in the controllers and directives is good or whether it pollutes that code too much. One idea is to catch errors in the action services, which will DRY up things and have a better guarantee that the error path is considered but it has the drawback of less flexibility in UI treatment, and I don't like opening modals in a service. Thoughts?

File: kifi-backend/modules/shoebox/angular/src/libraries/manageLibrary.js (hunk 3 above)
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/libraries/manageLibrary.js:85]: nice! :ice_cream:

**Comment by yipingliao** [comment]
Updated per your very helpful code review comments, @andrewconner and @ahsu1230

File: kifi-backend/modules/shoebox/angular/src/libraries/manageLibrary.js [hunk 4]
```diff
@@ -81,21 +81,18 @@ angular.module('kifi')
...
+                break;
+              case 'invalid_slug':
+                scope.$error.general = 'The URL you picked isn\'t valid. Try using only letters and numbers.';
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/libraries/manageLibrary.js:88]: One benefit of smart quotes is that they never have to be escaped!  :smiley: 
`“` ← `Opt` `[`
`”` ← `Shift` `Opt` `[`
`‘` ← `Opt` `]`
`’` ← `Shift` `Opt` `]`

Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/libraries/manageLibrary.js:88]: Oh! Thanks!

**Pull request merged**